src/components/AddBook.jsx

```diff
@@ -1,54 +1,53 @@
-import { useDispatch, useSelector } from 'react-redux';
+import { useDispatch } from 'react-redux';
 import { useState } from 'react';
-import { addBook } from '../redux/books/bookSlice';
+import { postBook } from '../redux/books/bookSlice';
 
 const AddNewBook = () => {
   const dispatch = useDispatch();
-  const [data, setData] = useState({
-    id: '-1', title: '', author: 'Susan Collins', category: 'Comedy',
-  });
-  const list = useSelector((state) => state.book.bookList);
-  const setTitle = (e) => {
-    setData((values) => ({
-      ...values,
-      title: e.target.value,
-      id: `${list.length + 1}`,
-    }));
-  };
-  const setAuthor = (e) => {
-    setData((values) => ({
-      ...values,
-      author: e.target.value,
-    }));
+
+  const [title, setTitle] = useState('');
+  const [author, setAuthor] = useState('');
+  const [category, setCategory] = useState('Comedy');
+
+  const saveTitle = (e) => {
+    e.preventDefault();
+    setTitle(e.target.value);
   };
 
-  const setCategory = (e) => {
-    setData((values) => ({
-      ...values,
-      category: e.target.value,
-    }));
+  const saveAuthor = (e) => {
+    e.preventDefault();
+    setAuthor(e.target.value);
   };
 
-  const submit = (e) => {
+  const saveCategory = (e) => {
     e.preventDefault();
-    dispatch(addBook(data));
+    setCategory(e.target.value);
+  };
+
+  const submit = () => {
+    if (title && author && category) {
+      try {
+        dispatch(postBook({ title, author, category }));
+      } catch (err) { console.log(err); }
+      document.getElementById('form').reset();
+    }
   };
```

Comment:
- [ ] 👋 Hey there!

#### 📝 Great work on this PR! I appreciate your effort in implementing the book submission feature. 📚✨

I have a couple of friendly suggestions to make the user experience even smoother:

- [ ] To prevent the page from refreshing when adding a book, let's make sure to remove the default form behavior by passing an event in the submit function and using `event.preventDefault()`. This will ensure a seamless book submission process without unnecessary page refreshes.

- [ ] Additionally, I noticed that when I added or deleted a book, I had to refresh the page to see the changes.  It's important that the state updates trigger automatic rendering, providing users with immediate feedback on their actions.

#### Keep up the good work! Let's make these enhancements for an even better user experience. 🚀💪 Thanks for your contribution!

Reply:
@frank1738 Thank you for the review. You are right, I have problems seeing the books, when I remove them. But I don't know how to solve it. Do you have any idea what is the problem?

Reply:
@BiancaMe .In the extraReducers section of your Redux slice, when postBook is fulfilled, it's common practice to update the bookList with the new book immediately so that the changes are reflected in your components without needing to wait for another API call to fetch the updated data. Here's how you can modify your postBook case to achieve this:

```
.addCase(postBook.fulfilled, (state, action) => {
  state.isLoading = false;
  // Create a new array with the existing books and the new book
  state.bookList = [...state.bookList, action.payload];
})


```

- [ ] you can also use the same approach when an action to remove a book is dispatched, you update the `bookList` Array

Reply:
@frank1738 Thank you for your help, I tried to do it but it doesn't work, show me this error:
![image](https://github.com/BiancaMe/bookstore-app/assets/37249375/59095264-0a89-491a-b05f-194076fc33da) .

Reply:
👍